presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java

```diff
@@ -497,7 +500,12 @@ else if (insertExistingPartitionsBehavior == InsertExistingPartitionsBehavior.ER
         if (!sortedBy.isEmpty()) {
             FileSystem fileSystem;
             try {
-                fileSystem = hdfsEnvironment.getFileSystem(session.getUser(), path, conf);
+                if (sortedWritingTempPath.isPresent()) {
+                    fileSystem = hdfsEnvironment.getFileSystem(session.getUser(), new Path(sortedWritingTempPath.get()), conf);
```

Comment:
If we could generate a path based on this property then we can use it for both fileSystem and when creating `SortedFileWriter` ?

`sortedWritingTempPath.map(Path::new).orElse(path)`

Reply:
Done